store/src/db/sql.rs

```diff
@@ -369,7 +376,8 @@ pub fn select_notes_since_block_by_tag_and_sender(
             note_hash,
             sender,
             tag,
-            merkle_path
+            merkle_path,
+            details
```

Comment:
This is probably OK for now, but in the future we should refactor this. The main reason is that `select_notes_since_block_by_tag_and_sender()` is used to build a `sync_state` response and for this response we don't need not details. With this approach, we'd be doing a lot of unnecessary work (e.g., reading the details form the database, parsing them etc.).

Let's create an issue to address this in the future.

Reply:
Should I just remove the `details` field from this query ?

Reply:
@phklive in public accounts we had the same issue and I removed this field from query and structure. I also added structure for specific endpoint for public accounts where this field is useful.

Reply:
To the extent of my understanding as of now I do not think that it would be wise to simply remove the details field and set all note details to `None` this would be misleading. Hence I prefer to open a separate issue to refactor this flow. 

Before opening the issue @bobbinth could you please clarify the reason why we would not need note details when syncing the state? I thought that we would also want to collect some new notes from the network using sync_state and that some of these notes could happen to be on-chain.

Reply:
> To the extent of my understanding as of now I do not think that it would be wise to simply remove the details field and set all note details to `None` this would be misleading. Hence I prefer to open a separate issue to refactor this flow.

Agreed. We should have a separate struct which is like note but without details (that's how it works for accounts, I believe). And yes, we can do it in a separate PR.

> Before opening the issue @bobbinth could you please clarify the reason why we would not need note details when syncing the state? I thought that we would also want to collect some new notes from the network using sync_state and that some of these notes could happen to be on-chain.

State sync could return many irrelevant notes (because we do a kind of "fuzzy" matching based on tags) and sending details for such notes could be expensive. So, the process is that with state sync we get minimal note info (Id + metadata + inclusion proofs) and if some of the notes happen to be on-chain, we can request details for them separately via the `GetNotesById` endpoint (this is actually one of the main reasons why we add this endpoint).

Reply:
Opened issue here: https://github.com/0xPolygonMiden/miden-node/issues/309

Reply:
Regarding this, with the current information we get from the sync, there is no way to tell whether the note is on chain or not, so we should add that information to the response.

Reply:
Does it not come with note metadata? If it is not there yet, we should make sure the protobuf messages include `note_type`.

Reply:
Correct, the sync endpoint specifically does not include the note type in any form.